src/shims/mod.rs

```diff
@@ -27,9 +28,74 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         // There are some more lang items we want to hook that CTFE does not hook (yet).
         if this.tcx.lang_items().align_offset_fn() == Some(instance.def.def_id()) {
-            // FIXME: return a real value in case the target allocation has an
-            // alignment bigger than the one requested.
-            let n = u128::max_value();
+            let ptr = this.read_scalar(args[0])?.not_undef()?;
+            let align = this.read_scalar(args[1])?.not_undef()?;
+
+            let n = {
+                let p = this.force_bits(ptr, this.pointer_size())?;
+                let a = this.force_bits(align, this.pointer_size())?;
+
+                let stride = this.memory().get(ptr.assert_ptr().alloc_id)?.align.bytes() as u128;
+                // if the allocation alignment is at least the required alignment, we use the
+                // libcore implementation
+                if stride >= a {
```

Comment:
we should have a mode if we are in nondeterministic mode, as we can just force_bits the pointer and run the libcore impl on the bare integer

Reply:
I can't parse this... "have a mode if we are in nondet mode"? There also is no det/nondet mode any more in Miri.

Reply:
Oh right... I forgot about that.

So yes, we can just unconditionally `force_bits` and run the libcore impl on the integers. Though as you said, the rest of the system may not be setup to handle it. If we pass libstd tests we should be ok enough though and can fix problems as they come

Reply:
I think we should only do any of this if the requested alignment is >= the allocation's alignment. That way, we can be sure that the rest of the system can handle this.